### DISCUSSION TITLE
setComponents - know if any problems with html

### DISCUSSION THREAD

**Discussion by BrendanMcDonagh-ContactFunnels** [GrapesJS/grapesjs#4872] [Q&A]
Hi,

Is there any way to know if the call to setComponents run without any errors? I noticed that if the data attribute of an html object element in an html string passed to setComponents contains a relative url (i.e <object type="image/svg+xml" data="images/main/offer.svg" class="uu">offer</object>) then my react application I am using with grapesjs will infinitely rerender and the files from my local server will infinitely be requested in a loop. I fixed this issue by converting the data attribute in object elements with absolute urls like I am doing with all other relative urls in html but I am expecting other html content in a string to cause this kind of issue and would like to know if there is a way of confirming success of call to setComponents?

Many thanks

**Comment by artf** [comment]
Well, `contains a relative url` it's not something the HTML parser might be aware of, as far the HTML is properly valid from the syntax point of view. 
I guess in such a case you should check the content on your own before passing it to something like `setComponents`
